### ISSUE TITLE
A11y semantics for some Material Chips should be "checkable"

### ISSUE THREAD

**Issue by nscobie** [flutter/flutter#60249]
Issue #58010 pointed out that `InputChip` was missing the `button: true` semantics.

PR #60141 applies the `button: true` semantics to all Material Chips (excluding the plain [info Chip](https://api.flutter.dev/flutter/material/Chip-class.html) which is non-interactive). That PR unblocks `customer: money (g3)` and brings our Chip semantics in-line with the behavior of the reference Material components [implementations for iOS](https://github.com/material-components/material-components-ios/blob/e1aee8556d04a64cef5149d51897f44711a5b077/components/Chips/tests/unit/ChipsTests.m#L100), but on Android _some_ Material Chips [have "checkable" semantics](https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/chip/Chip.java#L268) rather than button semantics, which is behavior that we should likely replicate (for `FilterChip` and `ChoiceChip`?) in the future.
